packages/hooks/use-popper/index.ts

```diff
@@ -0,0 +1,134 @@
+import { computed, onBeforeUnmount, ref, shallowRef, unref, watch } from 'vue'
+import { createPopper } from '@popperjs/core'
+
+import type { Ref } from 'vue'
+import type {
+  Instance,
+  Modifier,
+  Options,
+  State,
+  VirtualElement,
+} from '@popperjs/core'
+
+type ElementType = HTMLElement | undefined
+type ReferenceElement = ElementType | VirtualElement
+
+export const usePopper = (
+  referenceElementRef: Ref<ReferenceElement>,
+  popperElementRef: Ref<ElementType>,
+  opts: Ref<Options> | Options = {} as Options
+) => {
+  const stateUpdater = {
+    name: 'updateState',
+    enabled: true,
+    phase: 'write',
+    fn: ({ state }) => {
+      const elements = Object.keys(state.elements) as unknown as Array<
```

Comment:
The type casting here is mandatory since we are casting `string[]` to `(keyof State['element'])[]` here, it would not make sense to the compiler how both types were related, that is the reason behind `as unknown` here.